### PR TITLE
Add hotpot QA

### PR DESCRIPTION
**Pull request by ghomasHudson**
Added the [HotpotQA](https://github.com/hotpotqa/hotpot) multi-hop question answering dataset.

### CODE REVIEW

**Comment by lhoestq** [comment]
Awesome :) 

Don't pay attention to the RemoteDatasetTest error, I'm fixing it right now

File: datasets/hotpot_qa/hotpot_qa.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+@inproceedings{yang2018hotpotqa,
+  title={{HotpotQA}: A Dataset for Diverse, Explainable Multi-hop Question Answering},
+  author={Yang, Zhilin and Qi, Peng and Zhang, Saizheng and Bengio, Yoshua and Cohen, William W. and Salakhutdinov, Ruslan and Manning, Christopher D.},
+  booktitle={Conference on Empirical Methods in Natural Language Processing ({EMNLP})},
+  year={2018}
+}
+"""
+
+_DESCRIPTION = """\
+HotpotQA is a new dataset with 113k  Wikipedia-based question-answer  pairs with  four  key  features:  (1)  the  questions  require finding and reasoning over multiple supporting  documents  to  answer;  (2)  the  questions  are  diverse  and  not  constrained  to  any pre-existing  knowledge  bases  or  knowledge schemas;  (3)  we  provide  sentence-level  supporting facts required for reasoning, allowingQA systems to reason with strong supervisionand explain the predictions; (4) we offer a new type  of  factoid  comparison  questions  to  testQA  systems’  ability  to  extract  relevant  facts and perform necessary comparison.
```
Comment (lhoestq) [inline datasets/hotpot_qa/hotpot_qa.py:38]:
```suggestion
HotpotQA is a new dataset with 113k Wikipedia-based question-answer pairs with four key features:
(1) the questions require finding and reasoning over multiple supporting documents to answer;
(2) the questions are diverse and not constrained to any pre-existing knowledge bases or knowledge schemas;
(3) we provide sentence-level supporting facts required for reasoning, allowingQA systems to reason with strong supervisionand explain the predictions;
(4) we offer a new type of factoid comparison questions to testQA systems’ ability to extract relevant facts and perform necessary comparison.
```

I just removed spaces and added new lines for each one of the four points

**Review by lhoestq — COMMENTED** [review]
Good job thank you !
This dataset is a cool one

**Comment by lhoestq** [comment]
You can rebase from master to fix the CI test :)

**Comment by lhoestq** [comment]
If we're lucky we can even include this dataset in today's release

**Comment by ghomasHudson** [comment]
Just thinking since `type` can only be `comparison` or `bridge` and `level` can only be `easy`, `medium`, `hard` should they be `ClassLabel`?

**Comment by lhoestq** [comment]
> Just thinking since `type` can only be `comparison` or `bridge` and `level` can only be `easy`, `medium`, `hard` should they be `ClassLabel`?

I think it's more a tag than a label. I guess a string is fine

**Review by lhoestq — APPROVED** [review]
Looks all good thank you !

**Pull request merged**